sql/core/src/main/scala/org/apache/spark/sql/streaming/StreamingRepository.scala

```diff
@@ -0,0 +1,177 @@
+/*
+ * Changes for SnappyData data platform.
+ *
+ * Portions Copyright (c) 2017-2019 TIBCO Software Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package org.apache.spark.sql.streaming
+
+import java.text.SimpleDateFormat
+import java.util.{Date, TimeZone, UUID}
+
+import scala.collection.mutable.HashMap
+
+import org.apache.commons.collections.buffer.CircularFifoBuffer
+
+import org.apache.spark.ui.UIUtils
+
+class StreamingRepository {
+
+  val allQueries = HashMap.empty[UUID, StreamingQueryStatistics]
+
+}
+
+object StreamingRepository {
+
+  private val _instance: StreamingRepository = new StreamingRepository
+
+  def getInstance: StreamingRepository = _instance
+}
+
+
+class StreamingQueryStatistics (
+    qId: UUID,
+    qName: String,
+    runId: UUID,
+    startTime: Long,
+    trigger: Trigger = ProcessingTime(0L)) {
+
+  private val MAX_SAMPLE_SIZE = 100
```

Comment:
Done as suggested.
Added **_spark.sql.streaming.uiTrendsMaxSampleSize_** configuration parameter to configure how many historic data points be plotted on trends charts on structure streaming UI.
_**Note:**
In my opinion we should keep this parameter for internal use and let user not be aware of this config parameter. This is because Streaming UI charts are designed to fit in very small and limited space on UI page. Therefore large data sample if configured then charts will not be able to conveniently plot them._